### PR TITLE
Implement the changelog request and parsing in C++

### PR DESCRIPTION
**Pull request by suricactus**
This is an attempt to revive the changelog removed in https://github.com/opengisch/QField/pull/1375 , due to crashes on some devices, as described in https://github.com/opengisch/QField/issues/1327 and https://github.com/opengisch/QField/issues/1369 .

Now the implementation uses `QgsNetworkAccessManager` to connect to the outside world, after the lessons learned by making network requests from QML. 

sudo fairy make-apks

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available:
  - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-4ad115d093394d6659d720d67f79052982498219-arm64_v8a.apk)

Other architectures: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-4ad115d093394d6659d720d67f79052982498219-armv7.apk), [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-4ad115d093394d6659d720d67f79052982498219-x86_64.apk), [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-4ad115d093394d6659d720d67f79052982498219-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available:
  - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-93a74ffcc41fcc97ddc355c20ec74de08b0ba80f-arm64_v8a.apk)

Other architectures: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-93a74ffcc41fcc97ddc355c20ec74de08b0ba80f-armv7.apk), [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-93a74ffcc41fcc97ddc355c20ec74de08b0ba80f-x86_64.apk), [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-93a74ffcc41fcc97ddc355c20ec74de08b0ba80f-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available:
  - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-588a9833c620254dcc046c2bc11d5cd71e03e871-arm64_v8a.apk)

Other architectures: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-588a9833c620254dcc046c2bc11d5cd71e03e871-armv7.apk), [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-588a9833c620254dcc046c2bc11d5cd71e03e871-x86_64.apk), [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-588a9833c620254dcc046c2bc11d5cd71e03e871-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available:
  - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-36e468678cae83947417cb02e7859f92859a18de-arm64_v8a.apk)

Other architectures: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-36e468678cae83947417cb02e7859f92859a18de-armv7.apk), [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-36e468678cae83947417cb02e7859f92859a18de-x86_64.apk), [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-36e468678cae83947417cb02e7859f92859a18de-x86.apk)

File: src/core/changelogcontents.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class ChangelogContents : public QObject
+{
+    Q_OBJECT
+
+  public:
+    explicit ChangelogContents( QObject *parent = nullptr );
+
+    Q_INVOKABLE void request();
+
+    Q_INVOKABLE QString markdown();
```
Comment (m-kuhn) [inline src/core/changelogcontents.h:30]: Should we make this a readonly property?

File: src/core/changelogcontents.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+ChangelogContents::ChangelogContents( QObject *parent ):
+    QObject( parent )
+{
+}
+
+
+void ChangelogContents::request()
+{
+  QgsNetworkAccessManager *manager = new QgsNetworkAccessManager(this);
```
Comment (m-kuhn) [inline src/core/changelogcontents.cpp:32]: `QgsNetworkAccessManager::instance()`?

**Review by m-kuhn — COMMENTED** [review]
Nice

File: src/core/changelogcontents.cpp (hunk 2 above)
Comment (suricactus) [reply src/core/changelogcontents.cpp:32]: done!

**Comment by m-kuhn** [comment]
And finally the usual "some api docs please" request ;)

File: src/core/changelogcontents.h (hunk 1 above)
Comment (suricactus) [reply src/core/changelogcontents.h]: Hm, I did it the QML way, but I always get impressed it actually results into more code :)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available:
  - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-abae228347610fb69e8a470edcde3b9398db2ed1-arm64_v8a.apk)

Other architectures: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-abae228347610fb69e8a470edcde3b9398db2ed1-armv7.apk), [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-abae228347610fb69e8a470edcde3b9398db2ed1-x86_64.apk), [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-abae228347610fb69e8a470edcde3b9398db2ed1-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available:
  - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-59281cb493a7b807c94f9b4c61777ba5d341b0e7-arm64_v8a.apk)

Other architectures: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-59281cb493a7b807c94f9b4c61777ba5d341b0e7-armv7.apk), [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-59281cb493a7b807c94f9b4c61777ba5d341b0e7-x86_64.apk), [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-59281cb493a7b807c94f9b4c61777ba5d341b0e7-x86.apk)

File: src/qml/Changelog.qml [hunk 3]
```diff
@@ -6,149 +6,118 @@ import QtGraphicalEffects 1.0
...
+        color: '#95000000'
+        font: Theme.tipFont
+
+        fontSizeMode: Text.VerticalFit
+        textFormat: Text.MarkdownText
+        wrapMode: Text.WordWrap
+        Layout.fillWidth: true
+        Layout.fillHeight: true
+        Layout.minimumHeight: contentHeight
+        Layout.maximumHeight: contentHeight
```
Comment (m-kuhn) [inline src/qml/Changelog.qml:73]:
```suggestion
        Layout.maximumHeight: contentHeight
        visible: changelogContents.status != ChangelogContents.LoadingStatus
```

File: src/qml/Changelog.qml [hunk 4]
```diff
@@ -6,149 +6,118 @@ import QtGraphicalEffects 1.0
...
+              return changelogContents.markdown
+            case ChangelogContents.ErrorStatus:
+              return qsTr( 'Error while fetching changelog. Try again later.' )
+          }
+        }
 
-  function parseVersion(str) {
-    str = str.replace(/^[a-z]*/, '')
+        onLinkActivated: Qt.openUrlExternally(link)
+      }
```
Comment (m-kuhn) [inline src/qml/Changelog.qml:88]:
```suggestion
      }
      BusyIndicator {
          running: changelogContents.status == ChangelogContents.LoadingStatus
      }
```

**Review by m-kuhn — COMMENTED** [review]
I hope you don't mind further suggestion on irrelevant topics :laughing:

**Comment by suricactus** [comment]
:)
Peek 2020-11-20 00-07

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available:
  - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-b3aeb41e697e3854d1ac0abda7b0fcf44a9a2081-arm64_v8a.apk)

Other architectures: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-b3aeb41e697e3854d1ac0abda7b0fcf44a9a2081-armv7.apk), [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-b3aeb41e697e3854d1ac0abda7b0fcf44a9a2081-x86_64.apk), [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1424-b3aeb41e697e3854d1ac0abda7b0fcf44a9a2081-x86.apk)

**Review by m-kuhn — APPROVED** [review]

**Comment by nirvn** [comment]
Woupidou!

**Pull request merged**